packages/wallet-sdk/src/provider/CoinbaseWalletProvider.ts

```diff
@@ -804,6 +804,11 @@ export class CoinbaseWalletProvider extends EventEmitter implements Web3Provider
         return this._wallet_watchAsset(params);
     }
 
+    this.diagnostic?.log(EVENTS.METHOD_NOT_IMPLEMENTED, {
+      method: request.method,
+      sessionIdHash: this._relay ? Session.hash(this._relay.session.id) : undefined,
```

Comment:
following the existing pattern 
![image](https://github.com/coinbase/coinbase-wallet-sdk/assets/966662/784ba2fc-f982-4f25-ac15-a4b0be7867a5)